src/link.rs

```diff
@@ -32,34 +24,27 @@ pub fn process_auto_links(doc: &mut Spec) {
 }
 
 pub fn add_self_links(doc: &mut Spec) {
-    let dfn_els = match doc.dom().select("dfn") {
-        Ok(els) => els.map(|el| el.as_node().clone()).collect::<Vec<NodeRef>>(),
-        _ => Vec::new(),
-    };
-
-    if let Ok(heading_els) = doc.dom().select("h2, h3, h4, h5, h6") {
-        let mut found_first_numbered_section = false;
-
-        for heading_el in heading_els {
-            let heading_el = heading_el.as_node();
-
-            found_first_numbered_section |= html::get_attr(heading_el, "data-level").is_some();
-
-            if dfn_els.contains(heading_el) || !found_first_numbered_section {
-                continue;
-            }
-
-            // Append self-link.
-            if let Some(id) = html::get_attr(heading_el, "id") {
-                let a_el = html::new_a(
-                    btreemap! {
-                        "class" => "self-link".to_owned(),
-                        "href" => format!("#{}", id),
-                    },
-                    "",
-                );
-                heading_el.append(a_el);
-            }
+    let dfn_els = html::select(doc.dom(), "dfn").collect::<Vec<NodeRef>>();
+
+    let mut found_first_numbered_section = false;
+
+    for heading_el in html::select(doc.dom(), "h2, h3, h4, h5, h6") {
+        found_first_numbered_section |= html::get_attr(&heading_el, "data-level").is_some();
+
+        if dfn_els.contains(&heading_el) || !found_first_numbered_section {
```

Comment:
pre-existing, but I don't think this makes sense. How could `dfn_els` contain something that isn't a `dfn` (like here where you're iterating over `h{2,3,4,5,6}`?

Reply:
The "dfn" selector is actually `dfn, h2[data-dfn-type], h3[data-dfn-type], h4[data-dfn-type], h5[data-dfn-type], h6[data-dfn-type]`. I will fix that later.